### PR TITLE
Efs fixes for 5468 [CROM-6682]

### PR DESCRIPTION
**Pull request by nvanaja**
https://github.com/broadinstitute/cromwell/pull/6058

### CODE REVIEW

**Review by aednichols — APPROVED** [review]

File: supportedBackends/aws/src/main/scala/cromwell/backend/impl/aws/AwsBatchAsyncBackendJobExecutionActor.scala [hunk 1]
```diff
@@ -156,12 +161,18 @@ class AwsBatchAsyncBackendJobExecutionActor(override val standardParams: Standar
    * needs to push stuff out to S3. This is why we will eventually need
    * commandScriptContents here
    */
+
+  lazy val cmdScript =  configuration.fileSystem match {
```
Comment (kshakir) [inline supportedBackends/aws/src/main/scala/cromwell/backend/impl/aws/AwsBatchAsyncBackendJobExecutionActor.scala:165]: Any reason for the extra space?

File: supportedBackends/aws/src/main/scala/cromwell/backend/impl/aws/AwsBatchAsyncBackendJobExecutionActor.scala [hunk 2]
```diff
@@ -348,7 +359,14 @@ class AwsBatchAsyncBackendJobExecutionActor(override val standardParams: Standar
 
   // used by generateAwsBatchOutputs, could potentially move this def within that function
   private def generateAwsBatchSingleFileOutputs(womFile: WomSingleFile): List[AwsBatchFileOutput] = {
-    val destination = callRootPath.resolve(womFile.value.stripPrefix("/")).pathAsString
+     val destination = configuration.fileSystem match {
+      case  AWSBatchStorageSystems.s3 =>  callRootPath.resolve(womFile.value.stripPrefix("/")).pathAsString
+      case _ => DefaultPathBuilder.get(womFile.valueString) match {
+                 case p if !p.isAbsolute =>  callRootPath.resolve(womFile.value.stripPrefix("/")).pathAsString
+                 case p => p.pathAsString
```
Comment (kshakir) [inline supportedBackends/aws/src/main/scala/cromwell/backend/impl/aws/AwsBatchAsyncBackendJobExecutionActor.scala:366]: Are you open to using the same indentation used in the existing file, or did an editor suggest this?

File: supportedBackends/aws/src/main/scala/cromwell/backend/impl/aws/AwsBatchInitializationActor.scala [hunk 3]
```diff
@@ -69,6 +69,18 @@ class AwsBatchInitializationActor(params: AwsBatchInitializationActorParams)
   private val configuration = params.configuration
   override implicit val system = context.system
 
+  override def beforeAll(): Future[Option[BackendInitializationData]] = {
+    configuration.fileSystem match {
+     case AWSBatchStorageSystems.s3  => super.beforeAll
+     case _ => {
+                   initializationData map { data =>
+                   publishWorkflowRoot(data.workflowPaths.workflowRoot.pathAsString)
+                   Option(data)
```
Comment (kshakir) [inline supportedBackends/aws/src/main/scala/cromwell/backend/impl/aws/AwsBatchInitializationActor.scala:78]: A similar question about formatting, including whether the extra block inside the `case` statement is necessary?

File: supportedBackends/aws/src/test/scala/cromwell/backend/impl/aws/AwsBatchRuntimeAttributesSpec.scala [hunk 4]
```diff
@@ -352,7 +352,7 @@ class AwsBatchRuntimeAttributesSpec extends AnyWordSpecLike with CromwellTimeout
     val defaultedAttributes = RuntimeAttributeDefinition.addDefaultsToAttributes(
       staticRuntimeAttributeDefinitions, workflowOptions)(runtimeAttributes)
     val validatedRuntimeAttributes = runtimeAttributesBuilder.build(defaultedAttributes, NOPLogger.NOP_LOGGER)
-    AwsBatchRuntimeAttributes(validatedRuntimeAttributes, configuration.runtimeConfig)
+    AwsBatchRuntimeAttributes(validatedRuntimeAttributes, configuration.runtimeConfig,configuration.fileSystem)
```
Comment (kshakir) [inline supportedBackends/aws/src/test/scala/cromwell/backend/impl/aws/AwsBatchRuntimeAttributesSpec.scala:355]: Please add a space after the comma

**Review by kshakir — COMMENTED** [review]
Thanks for your code contribution while we all figure out our evolving processes and compliance procedures! Cromwell is a key part of the Broad's Data Sciences Platform, and DSP software needs to be covered by U.S. Federal authorizations along with private clinical lab audits too.

@kv076 can chime in with more clarification regarding the urgency of this PR. To be explicit for anyone following up later: as of Jan 19, unlike normal Broad DSP PRs, this one has zero regression/unit/integration/other tests to support the claim that this change provides a fix. Nor does it include tests to ensure that someone won't break this functionality again.

However, our various DSP teams have had multiple chats with @kv076 about testing code that will be included in Terra even if it is not exercised in Terra-prod, so we can leave it up to him on how to progress. There are numerous options for the future (plugin architectures, forked repositories, SDLC test requirements for all PRs, etc.), but afaik those decisions are still in the discussion and not the decision phase.

One easy fix, I believe for existing FedRAMP compliance/auditing every merged DSP PR description must refer to its accompanying JIRA ticket(s).

Someday too we'll use an automatic linter like the many other DSP code projects, but for now, I've put my 2¢ below.

**Comment by aednichols** [comment]
@kshakir we should file a ticket for this change but I think we’re good beyond that; its Terta impact description will accurately “none”

**Comment by nvanaja** [comment]
@kshakir Thanks for taking time to provide comments. Will checkin tests for the fix in a day or two. Sorry for not doing it earlier.

**Review by aednichols — APPROVED** [review]

**Comment by kshakir** [comment]
Hey @nvanaja-

Can you rebase this PR? Hopefully, there aren't any major conflicts since the main branch has continued to diverge.

Sorry, we have another set of (currently internal) CI tests that are failing on your branch. The team has [(also internally) discussed](https://broadinstitute.slack.com/archives/C1EH66VCM/p1612377943138400?thread_ts=1612377834.138300&cid=C1EH66VCM) the cryptic docker errors with `mysql-client`, and no one on the team has any quick concrete suggestions at this second other than rebasing. 🤞

Thanks again for your patience while we juggle your PR with our other work!

**Comment by nvanaja** [comment]
Hi @kshakir, 
Inadvertently did merge instead of rebase. Hope it's ok.
When building pull request, one of the jobs  failed first: https://api.travis-ci.com/v3/job/480472697/log.txt. Tried rerunning. It passed.
Would it be possible to share the internal failed test, I would like to run it locally in my workspace if possible. I can't access the failed test discussions link in your reply.
Thanks much.

**Review by kshakir — APPROVED** [review]
Status for anyone coming across this PR in the future:
- BT-57 removed the majority of backend testing designed to ensure that communication between cromwell and the cloud provider continues working as expected after future changes
- Therefore this PR was only tested on the two remaining tests (a hello world test in Travis, and one longer test in Jenkins) where neither existing test exercised the new features added in this PR
- This PR in the spirit of BT-57 cannot and does not introduce regression tests for the functionality, therefore if BT-57 is ever reverted then the tests for the functionality provided in this PR should also be added at that time
- This PR does verify via unit tests that any future code edits will still generate the same backend instruction syntax! 🥂

**Comment by kshakir** [comment]
@nvanaja

> Inadvertently did merge instead of rebase. Hope it's ok.

[It worked!](https://gotc-jenkins.dsp-techops.broadinstitute.org/job/cromwell-cron-aws/918/console) (still an internal-only link though)

> Tried rerunning. It passed.

Yeah... that's unfortunately cromwell's flaky tests. See BT-117 that's looking to help. Thanks for your patience and the retry.

> Would it be possible to share the internal failed test

No. Cromwell CI that runs under 3 hours is run on Travis CI. For over 3 hours cromwell uses the notoriously insecure Jenkins CI hosted by friends of ours in DSP. This was all set up before GitHub Actions existed. 🧓 Perhaps one day we'll move those tests where contributors like yourself can see the results, but afaik there are no current plans to do so.

> I would like to run it locally in my workspace if possible.

This is technically possible but isn't currently supported, and we'd have to all loop in our AppSec & DevOps folks to discuss further. In the short term hopefully you can put up with us as the folks-in-the-middle.

**Pull request merged**